core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FTPBatchSource.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.source;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.batch.BatchSource;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * {@link BatchSource} that reads from a FTP server
+ */
+@Plugin(type = "batchsource")
+@Name("FTP")
+@Description("Batch source for FTP source")
+public class FTPBatchSource extends FileBatchSource {
+  private static final Gson GSON = new Gson();
+  private static final Type MAP_STRING_STRING_TYPE = new TypeToken<Map<String, String>>() { }.getType();
+
+  @SuppressWarnings("unused")
+  private final FTPBatchSourceConfig config;
+
+  public FTPBatchSource(FTPBatchSourceConfig config) {
+    super(new FileBatchConfig(config.path, config.fileRegex, null, config.inputFormatClassName,
+                              limitSplits(config.fileSystemProperties), null));
+    this.config = config;
+  }
+
+  // Limit the number of splits to 1 since FTPInputStream does not support seek;
+  private static String limitSplits(@Nullable String fsProperties) {
+    Map<String, String> providedProperties;
+    if (fsProperties == null) {
+      providedProperties = new HashMap<>();
+    } else {
+      providedProperties = GSON.fromJson(fsProperties, MAP_STRING_STRING_TYPE);
+    }
+    providedProperties.put(FileInputFormat.SPLIT_MINSIZE, Long.toString(Long.MAX_VALUE));
```

Comment:
in the comment you say limiting the splits to 1, but the value is Long.MAX_VALUE, shouldn't it be 1 ?